presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestOrcFileRewriter.java

```diff
@@ -413,9 +428,225 @@ public void testUncompressedSize(boolean useOptimizedOrcWriter)
         }
 
         File newFile = new File(temporary, randomUUID().toString());
-        OrcFileInfo info = new OrcRecordFileRewriter().rewrite(file, newFile, new BitSet());
+        OrcFileInfo info = createFileRewriter(useOptimizedOrcWriter).rewrite(getColumnTypes(columnIds, columnTypes), file, newFile, new BitSet());
         assertEquals(info.getRowCount(), 3);
-        assertEquals(info.getUncompressedSize(), 55);
+        assertBetweenInclusive(info.getUncompressedSize(), 55L, 55L * 2);
+    }
+
+    /**
+     * The following test add or drop different columns
+     */
+    @Test(dataProvider = "useOptimizedOrcWriter")
```

Comment:
Add tests about the following cases:
- no column is left in the rewritten file
- all rows are deleted in the rewritten file
- all rows but the last one are deleted in the rewritten file

Reply:
`testRewriteAllRowsDeleted` will test the second case. I will add the other two.